### PR TITLE
Fix/replicasets left over

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - [x] ~Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?~
 - [ ] ~Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?~
 - [ ] ~Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?~
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

By default, k8s keeps `10` latest replica sets records.
Juju prefers no history records should be kept, so this PR sets it to `0`.

Drive by fix: validates `ScalePolicy` for non-stateful charms;

## QA steps

deploy a stateless charm then run upgrade-charm after changed k8s specs;

```console
# replicasets should always be one record per application;
$ mkubectl -n gitlab-k8s get rs -n t1
```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1864394

### CODE REVIEW

**Comment by ycliuhw** [comment]
[left over replicasets scaling k8s charm](https://trello.com/c/Y3e3wpsk/1060-left-over-replicasets-scaling-k8s-charm)

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -1090,6 +1090,12 @@ func (k *kubernetesClient) EnsureService(
 		}
 	}
 
+	if params.Deployment.DeploymentType != caas.DeploymentStateful {
+		if workloadSpec.Service != nil && workloadSpec.Service.ScalePolicy != "" {
+			return errors.NewNotValid(nil, fmt.Sprintf("ScalePolicy is only supported for %s application", caas.DeploymentStateful))
```
Comment (wallyworld) [inline caas/kubernetes/provider/k8s.go]: This check should also be done:
- at deploy time to reject an invalid deployment immediately
- when parsing and validating charm metadata (in charm.v6)

**Comment by ycliuhw** [comment]
!!build!!

File: caas/kubernetes/provider/k8s.go (hunk 1 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/k8s.go]: At that time, podspec isn't set yet.

Comment (wallyworld) [reply caas/kubernetes/provider/k8s.go]: ah right, sorry,  i had thought that was in the charm metadata

Comment (wallyworld) [inline caas/kubernetes/provider/k8s.go]: ... %s applications"

(missing "s")

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
!!build!!

**Comment by ycliuhw** [comment]
!!build!!

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**